languages/csharp/exercises/concept/object-initialization/.docs/after.md

```diff
@@ -0,0 +1,41 @@
+Object initializers are an alternative to constructors. The syntax is a comma separated list of field name=value pairs, illustrated below:
+
+```csharp
+public class Person
+{
+    public string Name;
+    public string Address;
+}
+
+...
+
+var person = new Person{Name="The President", Address = "Élysée Palace"};
+```
+
+Collections can also be initialized in this way. Typically, this is accomplished with comma separated lists as shown here:
+
+```csharp
+IList<Person> people = new List<Person>{ new Person(), new Person{Name="Joe", "Shmow"}};
+```
+
+Fields can be listed in any order or ommitted completely.
+
+Dictionaries use the following syntax:
+
+```csharp
+IDictionary<int, string> numbers = new Dictionary<int, string>{ [0] = "zero", [1] = "one"...};
+
+// or
+
+IDictionary<int, string> numbers = new Dictionary<int, string>{ {0, "zero }, {1,  "one"}...};
+```
+
+The initialized fields must be accessible to the caller. Typically, this means they must be `public` or `internal`.
+
+There is a strong trend promoting immutability amongst the language designers and leading practitioners, so the mutability of objects initialized in this way is seen as a disadvantage. There are proposals to introduce an `init` clause in the upcoming version of the language, C# 9, which may address this.
```

Comment:
A bit off-topic, but `init` properties felt like a really odd solution to me.

Reply:
Maybe a bit intrusive but it did address something (object initializers entail mutability) that bugged me.  

It must all seem a bit quaint from your F# perspective.

Reply:
> Maybe a bit intrusive but it did address something (object initializers entail mutability) that bugged me.

Absolutely agree. That always bugged me too, because I like both immutability _and_ object initializers.